contracts/TroveManager.sol

```diff
@@ -214,6 +214,18 @@ contract TroveManager is VestaBase, CheckContract, ITroveManager {
 			singleLiquidation.collGasCompensation
 		);
 
+		uint256 ICR = getCurrentICR(_asset, _borrower, vestaParams.priceFeed().fetchPrice(_asset));
```

Comment:
The ICR is already found in higher level functions. It will be better to use it as a function parameter rather to fetch the price again and calculate once more.

FetchPrice is not a simple get function and so cost a lot more gas that you might think.